src/duration.ts

```diff
@@ -0,0 +1,791 @@
+     */
+    public plusMicros(micros: number): Duration {
+        if (micros === 0) {
+            return this;
+        }
+
+        const seconds = intDiv(micros, LocalTime.MICROS_PER_SECOND);
+        const totalSeconds = addExact(this.seconds, seconds);
+
+        const remainder = micros % LocalTime.MICROS_PER_SECOND;
+        const nanos = this.nanos + (remainder * LocalTime.NANOS_PER_MICRO);
+
+        return Duration.ofSeconds(totalSeconds, nanos);
+    }
+
+    /**
+     * Returns a copy of this duration with the specified number of
+     * microseconds subtracted.
+     *
+     * @param micros - The microseconds to subtract
+     */
+    public minusMicros(micros: number): Duration {
+        return this.plusMicros(-micros);
+    }
+
+    /**
+     * Returns a copy of this duration with the specified number of nanoseconds
+     * added.
+     *
+     * @param nanos - The nanoseconds to add
+     */
+    public plusNanos(nanos: number): Duration {
+        if (nanos === 0) {
+            return this;
+        }
+
+        const seconds = intDiv(nanos, LocalTime.NANOS_PER_SECOND);
+        const totalSeconds = addExact(this.seconds, seconds);
+
+        const remainder = nanos % LocalTime.NANOS_PER_SECOND;
+        const newNanos = this.nanos + remainder;
+
+        return Duration.ofSeconds(totalSeconds, newNanos);
+    }
+
+    /**
+     * Returns a copy of this duration with the specified number of nanoseconds
+     * subtracted.
+     *
+     * @param nanos - The nanoseconds to subtract
+     */
+    public minusNanos(nanos: number): Duration {
+        return this.plusNanos(-nanos);
+    }
+
+    /**
+     * Checks if this duration is equal to another duration.
+     *
+     * @param other - The other duration
+     */
+    public equals(other: Duration): boolean {
+        return this.seconds === other.seconds && this.nanos === other.nanos;
+    }
+
+    /**
+     * Returns a copy of this duration multiplied by the specified multiplier.
+     *
+     * @param multiplier - The multiplier
+     */
+    public multipliedBy(multiplier: number): Duration {
+        if (multiplier === 0) {
+            return Duration.zero();
+        }
+
+        if (multiplier === 1) {
+            return this;
+        }
+
+        const seconds = multiplyExact(this.seconds, multiplier);
+        const nanos = multiplyExact(this.nanos, multiplier);
+
+        return Duration.ofSeconds(seconds, nanos);
+    }
+
+    /**
+     * Adds this duration to the specified instant.
+     *
+     * @param instant - The instant
+     */
+    public addTo(instant: Instant): Instant {
+        return instant.plusNanos(this.nanos).plusSeconds(this.seconds);
+    }
+
+    /**
+     * Subtracts this duration from the specified instant.
+     *
+     * @param instant - The instant
+     */
+    public subtractFrom(instant: Instant): Instant {
+        return instant.minusNanos(this.nanos).minusSeconds(this.seconds);
+    }
```

Comment:
Since we don't have a Temporal interface for simplicity, it's better to keep these methods inside their respective value object.